### ISSUE TITLE
Support PipeChannels that are half-closed to start with

### ISSUE THREAD

**Issue by weissi** [apple/swift-nio#1553]
This should be fairly straightforward to implement: We should have a `PipeChannel` that starts with only one file descriptor so is implicitly half-closed on the other one.

For more info see #288

**Comment by GeorgeLyon** [comment]
Just for posterity, it seems like `/dev/null` does not currently work on Linux where EPERM is raised during the `epoll_cfg` syscall. I've worked around this by opening a pipe and immedately closing the uninteresting end.

**Comment by weissi** [comment]
@GeorgeLyon Yes, you're right. On Linux you need to burn a pipe or something :| .

This actually seems to be the same issue as #2444

**Comment by Lukasa** [comment]
Dup of #2444.
